service/client/temporal/client.go

```diff
@@ -20,23 +21,45 @@ import (
 	"go.temporal.io/sdk/client"
 	"go.temporal.io/sdk/converter"
 	realtemporal "go.temporal.io/sdk/temporal"
+	"time"
 )
 
+type QueryWorkflowFailedRetryPolicy struct {
+	InitialIntervalSeconds int
+	MaximumAttempts        int
+}
+
 type temporalClient struct {
-	tClient        client.Client
-	namespace      string
-	dataConverter  converter.DataConverter
-	memoEncryption bool // this is a workaround for https://github.com/temporalio/sdk-go/issues/1045
+	tClient                        client.Client
+	namespace                      string
+	dataConverter                  converter.DataConverter
+	memoEncryption                 bool // this is a workaround for https://github.com/temporalio/sdk-go/issues/1045
+	queryWorkflowFailedRetryPolicy QueryWorkflowFailedRetryPolicy
 }
 
 func NewTemporalClient(
-	tClient client.Client, namespace string, dataConverter converter.DataConverter, memoEncryption bool,
+	tClient client.Client, namespace string, dataConverter converter.DataConverter, memoEncryption bool, retryPolicy *config.QueryWorkflowFailedRetryPolicy,
 ) uclient.UnifiedClient {
+	var rp QueryWorkflowFailedRetryPolicy
+
+	if retryPolicy.InitialIntervalSeconds == 0 {
```

Comment:
If I'm not mistaken, not providing `Config.ApiConfig.QueryWorkflowFailedRetryPolicy` in yaml file will make the default int value show (which is 0). In other words, we shouldn't need a nil check here